src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -1205,6 +1214,52 @@ class NPlusOneDBSpanDetectorExtended(NPlusOneDBSpanDetector):
     )
 
 
+class FileIOMainThreadDetector(PerformanceDetector):
+    """
+    Checks for a file io span on the main thread
+    """
+
+    __slots__ = ("spans_involved", "stored_problems")
+
+    settings_key = DetectorType.FILE_IO_MAIN_THREAD
+
+    def init(self):
+        self.spans_involved = {}
+        self.most_recent_start_time = {}
+        self.most_recent_hash = {}
+        self.stored_problems = {}
+
+    def visit_span(self, span: Span):
+        if self._is_file_io_on_main_thread(span):
+            settings_for_span = self.settings_for_span(span)
+            if not settings_for_span:
+                return
+
+            op, span_id, op_prefix, span_duration, settings = settings_for_span
+            if span_duration.total_seconds() * 1000 > settings["duration_threshold"]:
+                call_stack = ".".join(
+                    [item["function"] for item in span.get("data", {}).get("call_stack", [])]
+                ).encode("utf8")
+                hashed_stack = hashlib.sha1(call_stack).hexdigest()
+                fingerprint = f"1-{GroupType.PERFORMANCE_FILE_IO_MAIN_THREAD}-{hashed_stack}"
+                self.stored_problems[fingerprint] = PerformanceProblem(
+                    fingerprint=fingerprint,
+                    op=span.get("op"),
+                    desc=span.get("description", ""),
+                    parent_span_ids=[],
+                    type=GroupType.PERFORMANCE_FILE_IO_MAIN_THREAD,
+                    cause_span_ids=[span.get("span_id", None)],
```

Comment:
I think `cause_span_ids  should be empty in this case. "Cause" is for related spans that caused the bad spans, whereas `offender_span_ids` are the actual bad spans. (Cause might not make sense for anything other than N+1 tbh).